pwa/app/components/tba/charts/coprScatterChart.tsx

```diff
@@ -0,0 +1,296 @@
+import { useEffect, useState } from 'react';
+import {
+  Cell,
+  Label,
+  LabelList,
+  ReferenceLine,
+  Scatter,
+  ScatterChart,
+  Tooltip,
+  TooltipProps,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import {
+  NameType,
+  ValueType,
+} from 'recharts/types/component/DefaultTooltipContent';
+
+import { EventColors, TeamWithColor } from '~/api/colors';
+import type { EventCopRs } from '~/api/v3';
+import { Card, CardContent, CardHeader, CardTitle } from '~/components/ui/card';
+import { ChartContainer } from '~/components/ui/chart';
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectLabel,
+  SelectTrigger,
+  SelectValue,
+} from '~/components/ui/select';
+import { camelCaseToHumanReadable } from '~/lib/utils';
+
+interface Datapoint {
+  teamKey: string;
+  valueX: number;
+  valueY: number;
+}
+
+function generateWholeNumberedGridlinePoints(
```

Comment:
I'm not sure if these should be extracted to a utils file. I think we can do that if/when we need them again for another chart in the future, rather than this PR.